traffic_ops/traffic_ops_golang/auth.go

```diff
@@ -23,14 +23,15 @@ import (
 	"database/sql"
 
 	"github.com/apache/incubator-trafficcontrol/traffic_monitor_golang/common/log"
+	"github.com/jmoiron/sqlx"
```

Comment:
Developer's are error prone, and the cost of introducing this dependency is much less costly than the cost of consistent bug introduction

Reply:
Scan lines for large structs, 42 fields, could easily be improperly ordered, either when adding a new field or just pulling the object at some point. 
Using structs, even anonymous ones declared in the function when you don't need every field, with annotations mean you just need to verify the fields' db annotations are named properly.

Reply:
I agree it's better. It's just not worth the cost. Not convinced annotations are easier to get right than Scan, though.

Again, when-not-if the library stops being maintained, it'll require fixing every single endpoint. More likely, like Goose, we just won't do it, and we'll live with unmaintained and vulnerable code for years.

The performance is also an issue. It's not a huge slowdown, but it does use slow Reflection. If we keep making small choices against performance, we'll end up where we are with Perl via Death By A Thousand Cuts.

Reply:
> the cost of consistent bug introduction

> easily be improperly ordered

If your fields are improperly ordered, if they're the wrong types, it will immediately fail when run. Even if the types are correct, the values will be wrong, and immediately obvious when testing your code. But even if someone is pushing code without running it, we're requiring tests for every endpoint, right? Those tests should fail.

In short, I don't buy the "more bugs" argument, it would take a long cascade chain of failures, and complete lack of testing or tests, and pushing code without ever running it. You could make the same argument of any code.

Reply:
Do we know how much of a performance hit we're discussing? Can we toss together some quick benchmarks? Performance is pretty important here, but if it's a non-issue it'd be nice to know.

Reply:
@alficles Agree. Tangentially, we need to performance test Prepared queries too